pkg/distributor/distributor.go

```diff
@@ -494,6 +492,8 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 		}
 	}
 
+	tenantRetentionHours := d.tenantsRetention.RetentionHoursFor(tenantID, nil)
```

Comment:
I assume that in this case, we use the global tenant's retention period, rather than the retention period for the particular stream. 
If so, we can not do it because the metric that shows ingested bytes count always use the retention period from the  stream, and we need to follow it exactly

Reply:
I'll check but, we're only using `tenantRetentionHours` when we fail to parse the stream labels, how can the ingested_bytes know the stream retention period if it couldn't parse the stream labels?